internal/util/util.go

```diff
@@ -60,15 +65,51 @@ func GetTempDir() string {
 }
 
 // ExportFromReader reads bytes from given reader and exports to external tar, directory or stdout.
-func ExportFromReader(input io.Reader, opts define.BuildOutputOption) error {
+func ExportFromReader(input io.ReadCloser, store storage.Store, opts define.BuildOutputOption) error {
```

Comment:
The two ways to invoke this, with `input` and `store`, are _so different_, and the callers _must_ call them differently (we wouldn’t want the caller of the registry push to create the `ReadCloser` at all), that I think the two should clearly be separate functions.

Note: I didn’t check _at all_ whether this is the right place to, or the right way to, implement a push, e.g. by comparing the `buildah push` implementation.

Reply:
Tried moving the new push functionality to a separate function.

---

> Note: I didn’t check at all whether this is the right place to, or the right way to, implement a push, e.g. by comparing the buildah push implementation.

I took reference from buildah's push implementation for this, minus the additional meta[1].

[1] - https://github.com/danishprakash/buildah/blob/b5060e29537ec0283a948f59148a07f0c83d4e92/push.go#L101